File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/systemui/statusbar/network/NetworkSpeedSec.java
/*
  * This file is part of HyperCeiler.

  * HyperCeiler is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Affero General Public License as
  * published by the Free Software Foundation, either version 3 of the
  * License.

  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Affero General Public License for more details.

  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.

  * Copyright (C) 2023-2024 HyperCeiler Contributions
*/
package com.sevtinge.hyperceiler.module.hook.systemui.statusbar.network;

import com.sevtinge.hyperceiler.module.base.BaseHook;

public class NetworkSpeedSec extends BaseHook {
    @Override
    public void init() {
        try {
            findClass("com.android.systemui.statusbar.views.NetworkSpeedView").getDeclaredMethod("setNetworkSpeed", String.class);
            findAndHookMethod("com.android.systemui.statusbar.views.NetworkSpeedView",
                "setNetworkSpeed", String.class,
                new MethodHook() {
                    @Override
                    protected void before(MethodHookParam param) {
                        if (param.args[0] != null) {
                            String mText = (String) param.args[0];
                            param.args[0] = mText.replace("/", "")
                                .replace("s", "")
                                .replace("'", "")
                                .replace("วิ", "");
                        }
                    }
                }
            );
        } catch (NoSuchMethodException e) {
            try {
                findClass("com.android.systemui.statusbar.views.NetworkSpeedView").getDeclaredMethod("setNetworkSpeed", String.class, String.class);
                findAndHookMethod("com.android.systemui.statusbar.views.NetworkSpeedView",
                    "setNetworkSpeed", String.class, String.class,
                    new MethodHook() {
                        @Override
                        protected void before(MethodHookParam param) {
                            // logE(TAG, "1: " + param.args[0] + " 2: " + param.args[1]);
                            String mText = (String) param.args[1];
                            param.args[1] = mText.replace("/", "")
                                .replace("B", "")
                                .replace("s", "")
                                .replace("'", "")
                                .replace("วิ", "");
                        }
                    }
                );
            } catch (NoSuchMethodException f) {
                logE(TAG, "No such: " + e + " And: " + f);
            }
        }

    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/systemui/statusbar/NetworkSpeedSec.kt
package com.sevtinge.cemiuiler.module.systemui.statusbar

import com.github.kyuubiran.ezxhelper.utils.findMethod
import com.github.kyuubiran.ezxhelper.utils.hookBefore
import com.sevtinge.cemiuiler.module.base.BaseHook

//隐藏网速单位，仅供 Android12 及以上使用
object NetworkSpeedSec : BaseHook() {
    override fun init() {
        findMethod("com.android.systemui.statusbar.views.NetworkSpeedView") {
            name == "setNetworkSpeed" && parameterCount == 1
        }.hookBefore {
           if (mPrefsMap.getBoolean("hide_status_bar_network_speed_second")) {
                if (it.args[0] != null) {
                    val mText = (it.args[0] as String)
                        .replace("/", "")
                        .replace("s", "")
                        .replace("\'", "")
                        .replace("วิ", "")
                    it.args[0] = mText
                }
            }
        }
    }
}